enacts/wat_bal/maproom_monit.py

```diff
@@ -0,0 +1,512 @@
+    Input("submit_lat_lng","n_clicks"),
+    Input("map", "click_lat_lng"),
+    State("lat_input", "value"),
+    State("lng_input", "value")
+)
+def pick_location(n_clicks, click_lat_lng, latitude, longitude):
+    if dash.ctx.triggered_id == None:
+        lat = rr_mrg.precip["Y"][int(rr_mrg.precip["Y"].size/2)].values
+        lng = rr_mrg.precip["X"][int(rr_mrg.precip["X"].size/2)].values
+    else:
+        if dash.ctx.triggered_id == "map":
+            lat = click_lat_lng[0]
+            lng = click_lat_lng[1]
+        else:
+            lat = latitude
+            lng = longitude
+        try:
+            nearest_grid = pingrid.sel_snap(rr_mrg.precip, lat, lng)
+            lat = nearest_grid["Y"].values
+            lng = nearest_grid["X"].values
+        except KeyError:
+            lat = lat
+            lng = lng
+    return [lat, lng], lat, lng
+
+
+@APP.callback(
+    Output("wat_bal_plot", "figure"),
+    Input("loc_marker", "position"),
+    Input("map_choice", "value"),
+    Input("submit_kc", "n_clicks"),
+    State("planting_day", "value"),
+    State("planting_month", "value"),
+    State("crop_name", "value"),
+    State("kc_init", "value"),
+    State("kc_init_length", "value"),
+    State("kc_veg", "value"),
+    State("kc_veg_length", "value"),
+    State("kc_mid", "value"),
+    State("kc_mid_length", "value"),
+    State("kc_late", "value"),
+    State("kc_late_length", "value"),
+    State("kc_end", "value"),
+)
+def wat_bal_plots(
+    marker_pos,
+    map_choice,
+    n_clicks,
+    planting_day,
+    planting_month,
+    crop_name,
+    kc_init,
+    kc_init_length,
+    kc_veg,
+    kc_veg_length,
+    kc_mid,
+    kc_mid_length,
+    kc_late,
+    kc_late_length,
+    kc_end,
+):
+    lat = marker_pos[0]
+    lng = marker_pos[1]
+    kc_periods = pd.TimedeltaIndex(
+        [0, int(kc_init_length), int(kc_veg_length), int(kc_mid_length), int(kc_late_length)], unit="D"
+    )
+    kc_params = xr.DataArray(data=[
+        float(kc_init), float(kc_veg), float(kc_mid), float(kc_late), float(kc_end)
+    ], dims=["kc_periods"], coords=[kc_periods])
+    precip = rr_mrg.precip.isel({"T": slice(-366, None)})
+    p_d = precip["T"].where(
+        lambda x: (x.dt.day == int(planting_day))
+        & (x.dt.month == calc.strftimeb2int(planting_month)),
+        drop=True
+    ).squeeze(drop=True).rename("p_d")
+    precip = precip.where(precip["T"] >= p_d, drop=True)
+    try:
+        precip = pingrid.sel_snap(precip, lat, lng)
+        isnan = np.isnan(precip).sum().sum()
+        if isnan > 0:
+            error_fig = pingrid.error_fig(error_msg="Data missing at this location")
+            return error_fig
+    except KeyError:
+        error_fig = pingrid.error_fig(error_msg="Grid box out of data domain")
+        return error_fig
+    precip.load()
+    try:
+        sm, drainage, et_crop = ag.soil_plant_water_balance(
+            precip,
+            5,
+            60,
+            60./3.,
```

Comment:
I don't think we want them at the top of the file because they aim to be replaced asap by file reading that we don't want to do at top of files.
What is literal numbers in-line? like et=5
?

Reply:
OK if this is to be replaced soon then I don't mind so much.

Yes, by a literal I mean `5`, in contrast to a variable/constant `ET` that is set once and then used many times. If you need to change the value, it's easier to find all the occurrences of the variable `ET` than to find all the occurrences of the number 5 and try to figure out which of them are meant to be evapotranspiration constants and which represent some other quantity that happens also to be 5.